### ISSUE TITLE
What's the usage of "-ExpandProperties" parameter for Get-AzResource cmdlet?

### ISSUE THREAD

**Issue by imeya** [Azure/azure-powershell#11248]
I'm using Get-AzResource cmdlet with "-ExpandProperties" parameter, like below:

`Get-AzResource -ODataQuery "`$filter=resourcetype eq 'Microsoft.Web/sites'" -ExpandProperties`

But the result is same if I removed -ExpandProperties parameter, I wonder what's the usage of -ExpandProperties ? I read this [doc](https://docs.microsoft.com/en-us/powershell/module/az.resources/get-azresource?view=azps-3.5.0#parameters
), it says "When specified, expands the properties of the resource.". Not sure why the same result with/without this parameter.

**Comment by dingmeng-xue** [comment]
Involve Resource team to look into it. The example didn't describe it clearly.

**Comment by arpitjain099** [comment]
@imeya

I was facing the same issue earlier this morning. You can check out the example here (look for example 3):
https://docs.microsoft.com/ja-JP/powershell/module/azurerm.resources/get-azurermresource?view=azurermps-5.7.0#examples

Hope this solves your problem.

**Comment by imeya** [comment]
@arpitjain099 , thank you, I will take a look.

**Comment by pfeileon** [comment]
In my case `-ExpandProperties` throws an Error if I try to invoke `Get-AzResource` with a `ResourceGroupName`. `MissingSubscriptionFilter : The query string does not contain a 'SubscriptionId' filter.`
If I add `-Name *` it doesn't error but nothing is returned.
The main problem behind this is that it's not documented why querying a single resource returns a PSResource with a property `Properties` but querying multiple returns resources without it.
In conclusion, if you want to get the `Properties` property of all resources of a group, you need to get each resource separately, unless you specify the `ResourceType`.

**Comment by SaurabhSharma-MSFT** [comment]
@imeya Following up again to know if you still need help on this issue.

**Comment by imeya** [comment]
@SaurabhSharma-MSFT , thank you. it can be closed.

**Comment by SaurabhSharma-MSFT** [comment]
@imeya Thanks for the confirmation.

**Comment by jbla9028** [comment]
I was just reviewing how to resolve this issue and I'm not sure how to resolve. I am trying to run this command to get all the expanded properties of all resources in a single resource group. I can get around this by piping get-azresource to a select-object but I would assume if I have thousands of resources, this workflow would be extremely taxing and poor performance. How can I get expanded properties for resources just in 1 resource group without retrieving all resources first?

This is what works
$resources = Get-AzResource -ExpandProperties | Where-Object {$_.ResourceGroupName=$ResourceGroupName}

this is what I would expect to work

$resources = Get-AzResource -ExpandProperties -ResourceGroupName $ResourceGroupName}

**Comment by fgarcia-cnb** [comment]
a more efficiant workaround is to use an ODataQuery:

Get-AzResource -ODataQuery "`$filter=resourcegroup eq 'ResourceGroupName'" -ExpandProperties

**Comment by duncanmak** [comment]
I also found this to still be an issue, I used @fgarcia-cnb's solution, but like @jbla9028 wrote, it's fairly slow.

**Comment by duncanmak** [comment]
Talking to @msJinLei, this is still a bug.

**Comment by JFolberth** [comment]
Can confirm. I am running into this issue currently retrieving a new Databricks resources w/diagnostics configured. Az.Diagnostics does not return results yet the Invoke-AzRestMethod does.

**Comment by slarionoff** [comment]
Dear all,
for me this key doesn't work at all. Even when I run a command for a single resource by its name, the result fields set is the same for both runs (as well as those fields values themselves):
```
Get-AzResource -resourceId <Resource ID>
Get-AzResource -resourceId <Resource ID> -expandproperties

Result fields set:
Name              :
ResourceGroupName :
ResourceType      :
Location          :
ResourceId        :
Tags              :
```

**Comment by luigimalgas** [comment]
Has there been a solution for this as yet? Applying the filter on large resources is exceptionally slow. I had an implementation with Where-Object but it fails when now adding the -ExpandProperties

**Comment by fuixx217** [comment]
For anyone following, plus @luigimalgas - My workaround that seems fairly performant:
```Powershell
#looking for Arc-enabled SQL Servers and their properties.
$ResourceGroupName = "Arc"
$rg_sql_instances = @(Get-AzResource -ResourceGroupName $ResourceGroupName | Where-Object {$_.ResourceType -eq "microsoft.azurearcdata/sqlserverinstances" -and $_.Name -notlike "*SSIS*"}) | Get-AzResource -ExpandProperties
```

Basically, filter to resources in the RG without the `-ExpandProperties` specified, and then pipe that back to Get-AzResource with `-ExpandProperties`.  The result is what I was looking to obtain, and by limiting the resources to pipe by the RG and its where-object to filter for specific object types in the RG, I was able to limit the number of resources that ExpandProperties applied.

**Comment by fgarcia-cnb** [comment]
i recommend the ODataQuery sample i previously posted, since that utilizes server side filtering, which has much better performance. Filtering at the client side is much slower

**Comment by MaddyMicrosoft** [comment]
This has been fixed by #27368. `Get-AzResource -ExpandProperties` now handles resource group level and resource group plus name scenarios instead of falling back to the broken generic path. Closing as fixed. If a current `Get-AzResource -ExpandProperties` shape still fails, please open a fresh issue with the exact command and module version.
